plugins/arangodbendorsements/main.go

```
@@ -194,67 +186,82 @@ func (e *ArangoEndorsementStore) Init(args common.EndorsementStoreParams) error
+	var ok bool
+	var err error
 
-	ok := true
+	// Create an HTTP Connection First to the Client
 	arangoConn.conn, err = http.NewConnection(http.ConnectionConfig{
 		Endpoints: []string{"http://localhost:8529"},
 	})
 	if err != nil {
-		return false, fmt.Errorf("Failed to create HTTP connection: %v", err)
+		return fmt.Errorf("Failed to create HTTP connection: %v", err)
 	}
 
 	// Create a client
 	arangoConn.client, err = driver.NewClient(driver.ClientConfig{
 		Connection:     arangoConn.conn,
 		Authentication: driver.BasicAuthentication(e.Login, e.Password),
 	})
+	if err != nil {
+		return fmt.Errorf("Failed to create a client: %v", err)
+	}
 
-	// Check if Database with a given name exists or not?
+	// Check if Database with a given name exists or not
 	ok, err = arangoConn.client.DatabaseExists(ctx, e.StoreName)
-	if ok {
-		arangoConn.db, err = arangoConn.client.Database(ctx, e.StoreName)
-		if err != nil {
-			return false, fmt.Errorf("Failed to connect to Endorsement database: %v", err)
+	if err == nil {
+		if ok {
+			arangoConn.db, err = arangoConn.client.Database(ctx, e.StoreName)
+			if err != nil {
+				return fmt.Errorf("Failed to connect to the Endorsement database: %v", err)
+			}
 		}
+		// XXX(tho) -- what if !ok ?
+	} else {
+		return fmt.Errorf("Failure while checking whether DB exists: %v", err)
 	}
 
 	// Check if the Graph exists in the DB or not ?
 	ok, err = arangoConn.db.GraphExists(ctx, e.GraphName)
-	if ok {
-		// Connect to the Graph now to get the Graph
-		arangoConn.graph, err = arangoConn.db.Graph(ctx, e.GraphName)
-		if err != nil {
-			return false, fmt.Errorf("Failed to connect to Graph for Endorsement database: %v", err)
+	if err == nil {
+		if ok {
+			// Connect to the Graph now to get the Graph
+			arangoConn.graph, err = arangoConn.db.Graph(ctx, e.GraphName)
+			if err != nil {
+				return fmt.Errorf("Failed to connect to %s: %v", e.GraphName, err)
+			}
 		}
+		// XXX(tho) -- what if !ok ?
```

As this is the Verifier Plugin accessing the Arango Endorsements, Graph with a given name should always exist.
Any failure on Graph Connection should result in a failure and return. Hence please return a failure hereon line 241.